langroid/agent/base.py

```diff
@@ -690,6 +691,60 @@ def user_response(
                 ),
             )
 
+    async def user_response_async(
+        self,
+        msg: Optional[str | ChatDocument] = None,
+    ) -> Optional[ChatDocument]:
+        """
+        Asynch version of `user_response`. See there for details.
+        """
+        if not self.user_can_respond(msg):
+            return None
+
+        if self.callbacks.get_user_response_async is not None:
```

Comment:
As in the synchronous path, we need to skip when `default_user_response` is set.